be/src/vec/exec/file_scan_node.cpp

```diff
@@ -205,8 +231,9 @@ Status FileScanNode::close(RuntimeState* state) {
     _scan_finished.store(true);
     _queue_writer_cond.notify_all();
     _queue_reader_cond.notify_all();
-    for (int i = 0; i < _scanner_threads.size(); ++i) {
-        _scanner_threads[i].join();
+    {
+        std::unique_lock<std::mutex> l(_batch_queue_lock);
+        _queue_reader_cond.wait(l, [this] { return _num_running_scanners == 0; });
```

Comment:
Why not just using `_scanner_threads[i].join();`?

Reply:
Threads in the thread pool are reused and will not be terminated or released.